src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,138 @@
+     * @return the instantiated {@link BlockingSphereClient}.
+     */
+    protected static SphereClient createBlockingSphereClient(@Nonnull final SphereClientConfigOptions clientConfigOptions) {
+
+        final SphereClient underlyingClient = createSphereClient(clientConfigOptions.getClientConfig());
+        final SphereClient decoratedClient = decorateSphereClient(underlyingClient, clientConfigOptions.getMaxRetries(),
+                context -> calculateDurationWithExponentialRandomBackoff(context.getAttempt(),
+                        clientConfigOptions.getInitialDelay(), clientConfigOptions.getTimeout()), clientConfigOptions.getMaxParallelRequests());
+
+        return BlockingSphereClient.of(decoratedClient, clientConfigOptions.getTimeout(), clientConfigOptions.getTimeUnit());
+    }
+
+
+    protected static SphereClient createSphereClient(@Nonnull final SphereClientConfig clientConfig) {
+        final HttpClient httpClient = getHttpClient();
+        final SphereAccessTokenSupplier tokenSupplier =
+                SphereAccessTokenSupplier.ofAutoRefresh(clientConfig, httpClient, false);
+        return SphereClient.of(clientConfig, httpClient, tokenSupplier);
+    }
+
+    protected static SphereClient decorateSphereClient(
+            @Nonnull final SphereClient underlyingClient,
+            @Nonnull final long maxRetryAttempt,
+            @Nonnull final Function<RetryContext, Duration> durationFunction,
+            @Nonnull final int maxParallelRequests) {
+
+        final SphereClient retryClient = withRetry(underlyingClient, maxRetryAttempt, durationFunction);
+        return withLimitedParallelRequests(retryClient, maxParallelRequests);
+    }
+
+        /**
+         * Gets an asynchronous {@link HttpClient} to be used by the {@link BlockingSphereClient}.
+         *
+         * @return {@link HttpClient}
+         */
+    private static HttpClient getHttpClient() {
+        final AsyncHttpClient asyncHttpClient =
+                new DefaultAsyncHttpClient(
+                        new DefaultAsyncHttpClientConfig.Builder().build());
+        return AsyncHttpClientAdapter.of(asyncHttpClient);
+    }
+
+    private static SphereClient withRetry(
+            @Nonnull final SphereClient delegate,
+            long maxRetryAttempt,
+            @Nonnull final Function<RetryContext, Duration> durationFunction) {
+
+        final RetryAction scheduledRetry = RetryAction.ofScheduledRetry(maxRetryAttempt, durationFunction);
+        final RetryPredicate http5xxMatcher = RetryPredicate.ofMatchingStatusCodes(errCode -> errCode > 499 && errCode < 600);
+        final List<RetryRule> retryRules = Collections.singletonList(RetryRule.of(http5xxMatcher, scheduledRetry));
+        return RetrySphereClientDecorator.of(delegate, retryRules);
+    }
+
+    /**
+     * Computes a exponential backoff time delay in milliseconds to be used in retries, the delay grows with failed
```

Comment:
The method could be public, like a utility. 

Here the content of this Javadoc could make sense for all class, so you could create a good class Javadoc by using the content here with some udpates, actually, I could not see javadocs updates, the old class is not perfect so we could improve the all javadocs in here.